modules/nvidia_plugin/src/cuda/node_params.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2020-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cuda_runtime_api.h>
+
+#include <vector>
+
+namespace CUDA {
+
+struct NodeParams {
+    NodeParams(void* kernel, dim3 gridDim, dim3 blockDim) {
+        knp_.func = kernel;
+        knp_.gridDim = gridDim;
+        knp_.blockDim = blockDim;
+        knp_.sharedMemBytes = 0;
+        knp_.kernelParams = nullptr;
+        knp_.extra = nullptr;
+        ptrs_.reserve(20);
```

Comment:
reservation of vector space is redundant since number of arguments is unknown.

Reply:
I'd rather leave it to avoid little overhead in most cases.
Not reserving any space leads to vector reallocation one or several times even when 1st 5 elements are added/updated.